websocket-codec/src/message.rs

```diff
@@ -125,9 +128,26 @@ impl Message {
             None
         }
     }
+
+    /// For messages with opcode [`Opcode::Close`], returns the [`CloseFrame`].
+    /// Returns `None` otherwise.
+    pub fn as_close(&self) -> Option<CloseFrame> {
+        if matches!(self.opcode, Opcode::Close) && self.data.len() > 1 {
+            let mut code_arr = [0; 2];
+            code_arr.copy_from_slice(&self.data[..2]);
+            let code = u16::from_be_bytes(code_arr);
+            let reason = unsafe { str::from_utf8_unchecked(&self.data[2..]) };
+            Some(CloseFrame {
+                code: CloseCode::from(code),
+                reason: reason.to_string(),
```

Comment:
see above

Reply:
Line 139 - `self.data[2..]` needs to be checked for valid UTF-8 when the message is parsed and the message rejected if not, I don't see this check.